### PR TITLE
Streamline docs for new users

### PR DESCRIPTION
**Pull request by anbraten**
Reading through the docs using a predefined flow is extremely important especially for new-comers. Therefore this PR tries to align the docs so someone starting at the homepage could click their way through it based on the existing experience.

# TODO
- [x] improve overall getting started / intro (check the intent and experience level of the reader)
- [x] improve the getting started "first pipeline" section
- [x] improve the getting started "how to host my own wp instance"
- [x] remove unnecessary describing parts (how the workspace folder works, etc.) from the workflow syntax page

### CODE REVIEW

File: docs/blog/2024-01-01-continuous-deployment/index.md
[the remark this answers is not in the document]
Comment (anbraten) [reply docs/blog/2024-01-01-continuous-deployment/index.md:1]: Would like to leave it.

File: docs/docs/20-usage/10-intro.md [hunk 1]
```diff
@@ -1,73 +1,108 @@
...
+   + when:
+   +   - event: push
+   +     branch: main
 
-  - name: frontend
-    image: node:6
-    commands:
-      - npm install
-      - npm test
+   ...
```
[the remark this answers is not in the document]
Comment (anbraten) [reply docs/docs/20-usage/10-intro.md:46]: Might be more obvious using the current approach:

image

**Comment by 6543** [comment]
Wow this will make or docs even more accessible ❤️

File: docs/blog/2024-01-01-continuous-deployment/index.md
[the remark this answers is not in the document]
Comment (6543) [reply docs/blog/2024-01-01-continuous-deployment/index.md:1]: I think this needs future discusssion but for now it is good to go on my side.

I like the itterative aproach :)

**Pull request merged**